include/internal/safe_math.h

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_INTERNAL_SAFE_MATH_H
+# define OSSL_INTERNAL_SAFE_MATH_H
+# pragma once
+
+# include <openssl/e_os2.h>              /* For 'ossl_inline' */
+
+# ifndef OSSL_NO_OVERFLOW_CHECKING
```

Comment:
Would this be first OSSL_NO macro? Perhaps keeping OPENSSL_NO_.... is more aligned with the rest.

Reply:
The name isn't good, this is enabling the use of builtin overflow check functions.  It allows the C versions to be tested.  Alternatively, I could drop this and rely on the CIs to cover all paths (I'm not sure they do currently).

Reply:
Changed this to `OPENSSL_NO_BUILTIN_OVERFLOW_CHECKING` which is what it actually does.